app/models/team.rb

```diff
@@ -136,36 +119,41 @@ def score
               .pluck(:point_value, :'challenges.point_value').flatten.compact.sum
   end
 
-  def display_name
-    return self[:team_name] if eligible?
-
-    self[:team_name] + ' (ineligible)'
-  end
-
   # After remove callback passes a parameter which is the object that was just removed, we don't need it
   # so we just throw it away
-  def update_captain_and_eligibility(*)
+  def refresh_team_info(*)
     reload # SQL caching causing users to be empty when creating team making all teams ineligible
     set_team_captain
-    update_eligibility
-  end
-
-  # generates the body of the email certificate
-  def generate_certificate_body(doc, user_full_name, rank)
-    doc.bounding_box([55, 200], width: 640, height: 200) do
-      doc.font('Helvetica-Bold', size: 18) do
-        doc.text(I18n.t('users.team_completion_cert_string',
-                        full_name: user_full_name, team_name: team_name,
-                        score: score, rank: rank, team_size: division.teams.size),
-                 color: '005BA1', align: :center, leading: 4)
-      end
-    end
+    set_slots_available
+    set_eligibility
+    cleanup
   end
 
   private
 
   # If a team doesn't have a team captain but does have a user, set the team captain to the first user.
   def set_team_captain
-    update(team_captain: users.first) if team_captain.nil? && !users.empty?
+    if users.empty?
+      users << team_captain unless team_captain.nil?
+    elsif team_captain.nil?
+      update(team_captain: users.first)
+    end
+  end
+
+  # Number of users on team is calculated using game team size
+  def set_slots_available
+    updated_slots_available = division.game.team_size - users.count
```

Comment:
Saying `Game.instance` here is probably less error-prone and will use the cache. This will stop working if we ever add support for multiple games though.